lib/asciidoctor/converter/semantic-html5.rb

```diff
@@ -86,6 +91,35 @@ def generate_authors node
     end
   end
 
+  def generate_revision node
+    return unless (node.attr? 'revnumber') || (node.attr? 'revdate') || (node.attr? 'revremark')
+
+    revision_date = if (revdate = node.attr 'revdate')
+      date = ::Date._parse revdate
```

Comment:
One minor nitpick. It's not required that the revdate be a valid date. Date._parse already handles an invalid date gracefully. If the hash comes back empty, the datetime attribute should be dropped (or perhaps don't use the time element at all).

Reply:
I didn't known that it could return an empty hash. It can also return an incomplete hash!

```rb
p Date._parse("2eme janvier de l'an 511") # {:mon=>1, :mday=>2}
p Date._parse("l'an 511") # {:yday=>511}
p Date._parse('cinq cent onze av. J.-C.') # {}
```

> If the hash comes back empty, the datetime attribute should be dropped (or perhaps don't use the time element at all).

Since the `datetime` format is really flexible, we can still generate the datetime attribute if a value is missing.

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/time#valid_datetime_values

```html
<time datetime="01-02">2eme janvier de l'an 511</time>
```
```html
<time datetime="511">l'an 511</time>
```
```html
cinq cent onze av. J.-C.
```

Reply:
I'm fine with a partial date. But if the Hash it empty, I think it should be dropped.